### PR TITLE
fix correctness errors for AVS 2020-03-20 API

### PR DESCRIPTION
**Pull request by ctaggart**
This is to fix errors identified in:
- Swagger Correctness: MICROSOFT.AVS(Azure VMware Solution)
https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8187811
- linting error identified at https://aka.ms/LiveValidation.

The 11 errors that the CI check lists as  "breaking changes" fall into two categories:

- 1041 - AddedPropertyInResponse
These are all for Operations_List, to match what is actually returned.
- 1023 - TypeFormatChanged
The lint errors say that a format is required for integer. It was unspecified. They are all int32. I can roll back afa0274 if needed, but I'd prefer to fix this lint error.

The existing SDKs will continue to work against the 2020-03-20 API. The Python SDK reported one breaking change in its API for clusters.create_or_update signature.

---

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ctaggart** [comment]
The 11 errors listed as "breaking changes" listed fall into two categories:
- 1041 - AddedPropertyInResponse
  These are all for Operations_List, to match what is actually returned. 
- 1023 - TypeFormatChanged
  The lint errors says that a format is required for integer. It was unspecified. They are all int32. I can roll back afa02748920ca55b97781e4ee37e3b4e8bd6d248 if needed.

File: specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json [hunk 1]
```diff
@@ -1807,8 +1965,7 @@
         "nsxtPassword": {
           "description": "NSX-T Manager password",
           "type": "string",
-          "readOnly": true,
-          "x-ms-secret": true
```
Comment (ctaggart) [inline specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json:1811]: I need to roll this change back according to @pilor in https://github.com/Azure/azure-rest-api-specs/pull/10866#discussion_r507966702.

Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json:1811]: But CI fails with it in place, so I'm not sure what to do @ruowan.

Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json]: I put these back and suppressed the errors as recommended.

**Comment by ctaggart** [comment]
This is considered an urgent issue, so I created an ICM to get this reviewed:
https://portal.microsofticm.com/imp/v3/incidents/details/210561485/home

**Review by ruowan — APPROVED** [review]

**Comment by ctaggart** [comment]
@ruowan, there is a one single sdk breaking change reported by the pthon sdk check. It is saying:
- Operation ClustersOperations.create_or_update has a new signature

No other SDK has a breaking change. A stable version of the python SDK has *not* yet been published, so I do not think this is a blocking issue. If you look at https://pypi.org/project/azure-mgmt-avs/#history, you will see two versions:
- 1.0.0rc1 on July 14 for the 2020-03-20 API
- 0.1.0 on June 19 for the 2019-08-09 API

The az-vmware-cli uses autorust with --use=@microsoft.azure/autorest.python@3.0.62 and there are [no breaking changes](https://github.com/ctaggart/az-vmware-cli/pull/1). The breaking changes check is using --use=@microsoft.azure/autorest.python@~4.0.71. I'm not sure what version of autorust.python, the 1.0.0rc1 used.
 
I will ask for a stable 1.0.0 of azure-mgmt-avs after this is merged.

**Comment by ctaggart** [comment]
I created this breaking change review: https://msazure.visualstudio.com/One/_workitems/edit/8653604

**Comment by ruowan** [comment]
@changlong-liu Could you pls help to review this Python Breaking Change?

**Comment by changlong-liu** [comment]
Hi @ruowan @ctaggart , I added comments in the github issue: https://github.com/Azure/azure-sdk-for-python/issues/14735

Hi @ctaggart ,
for breaking change about create_or_update:
It related to the "Cluster" model definition changes in your PR. Originally there are only two properties in it is not readonly, (sku and clusterSize); currently there more properties become to non-readonly. This make the cluster parameter in create_or_update can be flattened in Python.

**Comment by ctaggart** [comment]
@changlong-liu those breaking changes are occurring when you run autorest against the current specification: https://github.com/ctaggart/azure-sdk-for-python/pull/1/files#diff-4db79dc989d60426c9b0788a2502c7c7c85acd73e15617a48794ca0691337080R179

There are no breaking changes caused by these spec changes in the Python SDK: https://github.com/ctaggart/azure-sdk-for-python/pull/2

**Comment by ctaggart** [comment]
@ruowan, the breaking changes review was approved. Please merge this. I'll request new SDK versions once merged.

**Pull request merged**